File: src/main/java/com/example/librarymanagement/exception/BookNotAvailableException.java
package com.example.librarymanagement.exception;

import org.springframework.http.HttpStatus;
import org.springframework.web.bind.annotation.ResponseStatus;

@ResponseStatus(HttpStatus.NOT_FOUND)
public class BookNotAvailableException extends RuntimeException {
    private Long libraryId;
    private Long bookId;

    public BookNotAvailableException(Long libraryId, Long bookId) {
        super(String.format("Book with id %d is not available in library with id %d", bookId, libraryId));
        this.libraryId = libraryId;
        this.bookId = bookId;
    }
}

File: src/main/java/com/example/librarymanagement/exception/BookNotAvailableException.kt
package com.example.librarymanagement.exception

import org.springframework.http.HttpStatus
import org.springframework.web.bind.annotation.ResponseStatus

@ResponseStatus(HttpStatus.NOT_FOUND)
class BookNotAvailableException(libraryId: String, bookId: String) : RuntimeException() {
    override val message: String = "Book with id $bookId is not available in library with id $libraryId"
}
